### PR TITLE
Feature Toggle - create component

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description

- [x] - create Rails component
- [ ] - ~create React component~ will handle in followup PR

### Screenshots

|  mobile  | desktop  |
|--------|--------|
|Screen Shot 2020-12-10 at 11 05 37 AM|Screen Shot 2020-12-10 at 11 05 12 AM|

### Spec
[Beta Features spec](https://www.figma.com/file/o8hxGSv1fEVPVm4oSF3c92/Beta-Features?node-id=0%3A1) 

### Steps for testing
1. http://localhost:4000/pages/object/feature_toggle and verify that component matches spec

### CODE REVIEW

File: docs/lib/sage_rails/app/views/sage_components/_sage_feature_toggle.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      type: "checkbox",
+      id: component.toggle_id,
+      hide_text: true,
+      label_text: component.title,
+      message: "",
+      name: component.toggle_name,
+      value: component.toggle_value,
+    } %>
+  </div>
+  <a class="sage-feature-toggle__img">
```
Comment (joedakroub) [inline docs/lib/sage_rails/app/views/sage_components/_sage_feature_toggle.html.erb:24]: Wondering if the link should be `sage-feature-toggle__link` and the image `sage-feature-toggle__image`?

File: docs/app/views/examples/objects/feature_toggle/_preview.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  <%= sage_component SageFeatureToggle, {
+    description: "Quickly see Segments & filter contacts with a consistent People list view.",
+    image: asset_pack_path("media/images/docs/card/card-placeholder-sm.png"), 
+    link_text: "Learn more",
+    link_location: "#",
+    title: "New People list view",
+    toggle_id: "feature-1",
+    toggle_name: "feature-1",
+    toggle_value: "feature 1",
+  } %>
```
Comment (joedakroub) [inline docs/app/views/examples/objects/feature_toggle/_preview.html.erb:11]: Another option may be to group the `link` and `toggle` props together?

```
link: {
  text: "",
  location: ""
},
toggle: {
  id: "",
  name: "",
  value: ""
}
```

**Review by joedakroub — APPROVED** [review]
Looking really good Quinton! Added a couple of comments, but nothing blocking. 👍

File: docs/app/views/examples/objects/feature_toggle/_preview.html.erb (hunk 2 above)
Comment (philschanely) [reply docs/app/views/examples/objects/feature_toggle/_preview.html.erb:11]: I'm in favor of this kind of structuring too. Only ❓ is how to also indicate such a structure in the Rails type declarations. The flat structure may be best for now.

File: docs/lib/sage_rails/app/views/sage_components/_sage_feature_toggle.html.erb (hunk 1 above)
Comment (philschanely) [reply docs/lib/sage_rails/app/views/sage_components/_sage_feature_toggle.html.erb:24]: ... or perhaps `sage-feature-toggle__image-link` and `__image`?

File: packages/sage-assets/lib/stylesheets/patterns/objects/_feature_toggle.scss [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+.sage-feature-toggle {
+  display: grid;
+  grid-column-gap: sage-spacing(md);
+  grid-row-gap: sage-spacing(sm);
+  
+  @media (max-width: sage-breakpoint(md-min)) {
+    grid-template-columns: 1fr min-content;
+    grid-template-rows: $-image-height-mobile auto auto;
+    grid-template-areas:
+      "img aside"
```
Comment (philschanely) [inline packages/sage-assets/lib/stylesheets/patterns/objects/_feature_toggle.scss:21]: What do you think about whitespace that makes the cells clearer?

```
"img     aside"
"title   title"
"content content"
```

**Review by philschanely — APPROVED** [review]
Nothing blocking but a few questions/comments. Nice job on this!! 🏆

**Pull request merged**